files/external_node_v2.rb

```diff
@@ -163,6 +165,8 @@ def generate_fact_request(certname, filename)
     rescue => e
       raise "Could not generate facts for Foreman: #{e}"
     end
+  else
+    $stderr.puts 'Last facts are older than last upload time, rerun puppet agent on host to get newer facts'
```

Comment:
Nope, that was when running `puppet agent --test` or even on normal Puppet runs:
```
# grep -c 'rerun puppet agent on host to get newer facts' puppetserver.log
149
```
That's from the last 10 hours or so.